src/main/java/ch/uzh/ifi/hase/soprafs23/service/GroupService.java

```diff
@@ -65,23 +64,20 @@ public Group getGroupById(Long id) {
         return group.get();
     }
 
+    // TODO: why does this exist? it doesn't flush the repo, so it wouldn't work, but it's also never used
     public Group updateGroup(Long groupId, String newGroupName) {
-        Group groupToUpdate = groupRepository.findById(groupId)
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Group not found"));
+        Group groupToUpdate = getGroupById(groupId);
         groupToUpdate.setGroupName(newGroupName);
         return groupRepository.save(groupToUpdate);
     }
 
-
+    // TODO: why does this exist? it's even tested! but only used where it's tested nowhere else
```

Comment:
This was made by me bc it was part of one of my dev tasks. I did it for future use --> thats why it is tested as well.

Reply:
Imo it can stay and we use it for checking whether after allowing for drop outs, there is still at least one guest within the group